connector-email/src/main/java/com/redhat/cloud/notifications/connector/email/processors/mbop/MBOPRequestPreparer.java

```diff
@@ -0,0 +1,88 @@
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+@ApplicationScoped
+public class MBOPRequestPreparer implements Processor {
+
+    @ConfigProperty(name = "notifications.connector.user-provider.bop.api_token")
+    String bopApiToken;
+
+    @ConfigProperty(name = "notifications.connector.user-provider.bop.client_id")
+    String bopClientId;
+
+    @ConfigProperty(name = "notifications.connector.user-provider.bop.env")
+    String bopEnv;
+
+    @ConfigProperty(name = "notifications.connector.email.no-reply.address")
+    String noReplyEmail;
+
+    @Inject
+    FeatureFlipper featureFlipper;
+
+    /**
+     * Prepares the payload that BOP/MBOP expects.
+     * @param exchange the exchange of the pipeline.
+     */
+    @Override
+    public void process(final Exchange exchange) {
+        final String subject = exchange.getProperty(ExchangeProperty.RENDERED_SUBJECT, String.class);
+        final String body = exchange.getProperty(ExchangeProperty.RENDERED_BODY, String.class);
+        final Set<User> recipients = exchange.getProperty(ExchangeProperty.USERS, Set.class);
+
+        final Email email = this.buildEmail(recipients, subject, body);
+
+        final Emails emails = new Emails();
+        emails.addEmail(email);
+
+        // Specify the message's payload in JSON.
+        exchange.getMessage().setBody(JsonObject.mapFrom(emails));
+
+        // Specify the request's method.
+        exchange.getMessage().setHeader(Exchange.HTTP_METHOD, HttpMethods.POST);
+
+        // Specify the payload's content type.
+        exchange.getMessage().setHeader(Exchange.CONTENT_TYPE, "application/json");
+
+        // Specify the authentication details required by MBOP/BOP.
+        exchange.getMessage().setHeader(Constants.BOP_API_TOKEN_HEADER, this.bopApiToken);
+        exchange.getMessage().setHeader(Constants.BOP_CLIENT_ID_HEADER, this.bopClientId);
+        exchange.getMessage().setHeader(Constants.BOP_ENV_HEADER, this.bopEnv);
+    }
+
+    /**
+     * Builds the email payload.
+     * @param recipients the set of filtered recipients.
+     * @param subject the rendered subject of the email.
+     * @param body the rendered body of the email.
+     * @return an HTML {@link Email} instance with the provided data. If the
+     * default recipient is enabled, then that will be used for the recipient.
+     */
+    protected Email buildEmail(final Set<User> recipients, final String subject, final String body) {
+        final Set<String> usersEmail = recipients.stream().map(User::getUsername).collect(Collectors.toSet());
+
+        final Email email = new Email();
+        email.setBodyType(com.redhat.cloud.notifications.connector.email.constants.Email.EMAIL_BODY_TYPE_HTML);
+        if (this.featureFlipper.isAddDefaultRecipientOnSingleEmail()) {
+            email.setRecipients(Set.of(this.noReplyEmail));
+        }
```

Comment:
This will most likely be dropped soon in the `engine`. I wonder if we should simply remove the noreply email entirely from the connector.

Reply:
Okay, completely removed this feature